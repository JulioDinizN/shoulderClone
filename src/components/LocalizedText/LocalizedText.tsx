import {Text} from 'react-native';
import {translate} from '../../localization/services';

/**
 * what could be a FUCKING AWESOME COMPONENT, must have:
 * - typing
 * - validation
 * - default style or PROPERTY that are optional
 * - documentation with all properties and describing the respoibility of the component
 * - tests (minimum render with validation of properties [defined/not null, etc])
 * - storybook with the component and its own states.
 *
 */

// always add what kind of component it is in the begning of the proptype
// because can  be used in other components. e.g: organism
export type LocalizedTextProps = {
  textKey: string;
  interpolation?: Object;
  style?: Object;
};

/**
 * A component that returns JSX with the translation key resolved. Supports interporlation of dynamic keys.
 * @param {string} textKey e.g: textHomeScreenTitle
 * @param {string=} interpolation  dynamic key to be added to the component.
 * e.g <LocalizedText textKey="homeScreenTitle" interpolation={myVariableToAddToText}>
 * @returns { Component } with the translation key resolved.
 *
 */
export const LocalizedText = ({
  textKey,
  interpolation = {},
  style = {},
  ...rest
}: LocalizedTextProps) => {
  return (
    <Text style={style} {...rest}>
      {translate(textKey, interpolation)}
    </Text>
  );
};
