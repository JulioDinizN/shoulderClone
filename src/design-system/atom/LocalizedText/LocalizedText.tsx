import {Text} from 'react-native';


/**
 * what could be a FUCKING AWESOME COMPONENT, must have:
 * - typing
 * - validation
 * - default style or PROPERTY that are optional
 * - documentation with all properties and describing the respoibility of the component
 * - tests (minimum render with validation of properties [defined/not null, etc])
 * - storybook with the component and its own states.
 * 
*/



// always add what kind of component it is in the begning of the proptype
// because can  be used in other components. e.g: organism
export type LocalizedTextProps = {
  textKey: string;
  interpolation?: Object;
  style?: Object;
};

/**
 * A component that returns JSX with the translation key resolved. Supports interporlation of dynamic keys.
 * @param textKey {string} e.g: textHomeScreenTitle 
 * @param {interpolation?} <string> dynamic key to be added to the component. 
 * e.g <LocalizedText textKey="homeScreenTitle" interpolation={myVariableToAddToText}>
 * @returns { Component } with the translation key resolved.
 * 
 */
export const LocalizedText = ({
  textKey,
  interpolation = {},
  style = {},
  ...rest
}: LocalizedTextProps) => {
 
  // if (!_.notDefined((textKey)) && object vazio e array vazio, tem q ser valido) {
  //   throw new Error("Missing textKey that must be of type = <string>")
  // }

  const textToShow = 'hello';

  return (
    <Text style={style} {...rest}>
      {textToShow}
    </Text>
  );
};


// default property here
