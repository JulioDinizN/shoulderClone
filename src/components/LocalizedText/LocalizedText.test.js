import renderer from 'react-test-renderer';
import { LocalizedText } from './LocalizedText'


describe('Localized Text Component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<LocalizedText />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})


