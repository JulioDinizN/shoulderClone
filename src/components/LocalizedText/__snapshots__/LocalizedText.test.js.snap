// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Localized Text Component should render correctly 1`] = `
<Text
  style={Object {}}
/>
`;
